Flaky crash in tests on Windows

e.g. https://github.com/tconbeer/harlequin/actions/runs/13146050150/job/36684296449

trace:
<details>
<pre>
================================== FAILURES ===================================
_________________________ test_harlequin_loads_cache __________________________

cache = Cache(focus_index=1, buffers=[BufferState(selection=Selection(start=(0, 3), end=(0, 3)), text='select 1\n'), BufferSta... 0), end=(0, 0)), text=''), BufferState(selection=Selection(start=(0, 0), end=(1, 0)), text='select\n*\nfrom\nfoo\n')])
app = Harlequin(title='Harlequin', classes={'-dark-mode'}, pseudo_classes={'focus', 'dark'})

    @pytest.mark.use_cache
    @pytest.mark.asyncio
    async def test_harlequin_loads_cache(cache: Cache, app: Harlequin) -> None:
        write_cache(cache)
>       async with app.run_test() as pilot:

tests\functional_tests\test_cache.py:64: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\contextlib.py:206: in __aexit__
    await anext(self.gen)
.venv\lib\site-packages\textual\app.py:1933: in run_test
    raise self._exception
.venv\lib\site-packages\textual\message_pump.py:609: in _process_messages_loop
    await self._dispatch_message(message)
.venv\lib\site-packages\textual\message_pump.py:685: in _dispatch_message
    await self._on_message(message)
.venv\lib\site-packages\textual\message_pump.py:775: in _on_message
    await invoke(method, message)
.venv\lib\site-packages\textual\_callback.py:93: in invoke
    return await _invoke(callback, *params)
.venv\lib\site-packages\textual\_callback.py:53: in _invoke
    result = callback(*params[:parameter_count])
src\harlequin\app.py:473: in update_internal_editor_state
    self.editor = self.editor_collection.current_editor
src\harlequin\components\code_editor.py:177: in current_editor
    content = self.query_one(ContentSwitcher)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = EditorCollection()
selector = <class 'textual.widgets._content_switcher.ContentSwitcher'>
expect_type = None

    def query_one(
        self,
        selector: str | type[QueryType],
        expect_type: type[QueryType] | None = None,
    ) -> QueryType | Widget:
        """Get a widget from this widget's children that matches a selector or widget type.
    
        Args:
            selector: A selector or widget type.
            expect_type: Require the object be of the supplied type, or None for any type.
    
        Raises:
            WrongType: If the wrong type was found.
            NoMatches: If no node matches the query.
    
        Returns:
            A widget matching the selector.
        """
        _rich_traceback_omit = True
    
        if isinstance(selector, str):
            query_selector = selector
        else:
            query_selector = selector.__name__
    
        try:
            selector_set = parse_selectors(query_selector)
        except TokenError:
            raise InvalidQueryFormat(
                f"Unable to parse {query_selector!r} as a query; check for syntax errors"
            ) from None
    
        if all(selectors.is_simple for selectors in selector_set):
            cache_key = (self._nodes._updates, query_selector, expect_type)
            cached_result = self._query_one_cache.get(cache_key)
            if cached_result is not None:
                return cached_result
        else:
            cache_key = None
    
        for node in walk_depth_first(self, with_root=False):
            if not match(selector_set, node):
                continue
            if expect_type is not None and not isinstance(node, expect_type):
                continue
            if cache_key is not None:
                self._query_one_cache[cache_key] = node
            return node
    
>       raise NoMatches(f"No nodes match {selector!r} on {self!r}")
E       textual.css.query.NoMatches: No nodes match <class 'textual.widgets._content_switcher.ContentSwitcher'> on EditorCollection()

.venv\lib\site-packages\textual\dom.py:1473: NoMatches
--------------------------- snapshot report summary ---------------------------
114 snapshots passed.
=========================== short test summary info ===========================
FAILED tests/functional_tests/test_cache.py::test_harlequin_loads_cache - textual.css.query.NoMatches: No nodes match <class 'textual.widgets._content_switcher.ContentSwitcher'> on EditorCollection()
============ 1 failed, 169 passed, 7 skipped in 1[45](https://github.com/tconbeer/harlequin/actions/runs/13146050150/job/36684296449#step:8:46).96s (0:02:25) =============
</pre>
<details>

--- a/tests/functional_tests/test_query_editor.py
+++ b/tests/functional_tests/test_query_editor.py
@@ -211,8 +211,10 @@ async def test_member_autocomplete(
             await pilot.pause()
         for db_node in app.data_catalog.database_tree.root.children:
             await _expand_and_wait(db_node)
+            await wait_for_workers(app)
             for schema_node in db_node.children:
                 await _expand_and_wait(schema_node)
+                await wait_for_workers(app)
         await pilot.pause(1)
 
         # now the completer should be populated

